src/Analyzers/CSharp/Analyzers/MakeLocalFunctionStatic/MakeLocalFunctionStaticHelper.cs

```diff
@@ -17,23 +18,46 @@ internal static class MakeLocalFunctionStaticHelper
         public static bool IsStaticLocalFunctionSupported(SyntaxTree tree)
             => tree.Options is CSharpParseOptions csharpOption && csharpOption.LanguageVersion >= LanguageVersion.CSharp8;
 
-        public static bool TryGetCaputuredSymbols(LocalFunctionStatementSyntax localFunction, SemanticModel semanticModel, out ImmutableArray<ISymbol> captures)
+        private static bool TryGetDataFlowAnalysis(LocalFunctionStatementSyntax localFunction, SemanticModel semanticModel, [NotNullWhen(returnValue: true)] out DataFlowAnalysis? dataFlow)
         {
-            var dataFlow = semanticModel.AnalyzeDataFlow(localFunction);
-            if (dataFlow is null)
+            dataFlow = semanticModel.AnalyzeDataFlow(localFunction);
+            return dataFlow is { Succeeded: true };
+        }
+
+        private static bool CanBeCalledFromStaticContext(LocalFunctionStatementSyntax localFunction, DataFlowAnalysis dataFlow)
+        {
+            // If other local functions are called the it can't be made static unles the
+            // are static, or the local function is recursive, or its calling a child local function
+            return !dataFlow.UsedLocalFunctions.Any(lf => !lf.IsStatic && !IsChildOrSelf(localFunction, lf));
```

Comment:
>lf => !lf.IsStatic && !IsChildOrSelf(localFunction, lf) [](start = 52, length = 55)

Could mark this lambda `static`

Reply:
Err, no I can't, it captures `localFunction`. Unless i'm misunderstanding?

Reply:
No, I missed that.